ls1intum/Artemis
# `Development`: Add auxiliary repository support for Local VC CI

### Checklist
#### General


#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).(https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
#### Client


#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 11 (Local VC CI).
Gitlab).

### Motivation and Context


The current Local VC/CI environment only partially supports auxiliary repositories. While auxiliary repositories theoretically can be created in the local VCS, they can neither be accessed by anyone (the Local VC service does not check for auxiliary repos and thus authorization fails) nor are they added to the build plan.

### Description
This PR adds the possibilty to create, access and push to auxiliary repositories. Auxiliary repositories (if they exist for an exercise) are bound to the build job container when a build job is triggered. Within the build container, they are checked out to the specified directories.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Tutor
- 1 Students
- 1 Programming Exercise with at least one auxiliary repository
- Testserver 11

1. Log in as Instructor
  i. Clone aux repo (should work)
  ii. Push to aux repo (should work)
2. Log in as Tutor
  i. Clone aux repo (should work)
  ii. Push to aux repo (should **not** work)
3. Log in as Student 
  i. Clone aux repo (you can use the URL from Instructor/Tutor and change the identifier) (should **not** work)
  ii. Push to aux repo (should **not** work)
  iii. Create new submission (should work as expected)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
